### DISCUSSION TITLE
Does duckdb load the entire dataset into memory?

### DISCUSSION THREAD

**Discussion by siddhsql** [duckdb/duckdb#8158] [General]
I understand duckdb is an embedded database but is it also an in-memory database? does it load the entire dataset into memory?

**Comment by yoonghm** [comment]
I understand that if it runs as in-memory then the entire database would be in the computer memory. However, it the database is backed by a physical file then the in-process database could be larger than the available memory. 

However, I am not clear is that operating systems, utilizing demand paging, allow a process to access memory within its addressable location but is above it'd physical address.
